File: Cc.Core/Models/Company.cs
using System;
using System.Collections.Generic;

namespace Cc.Core.Models;

public partial class Company
{
    public long Id { get; set; }

    public string BusinessName { get; set; } = null!;

    public string Address { get; set; } = null!;

    public string VatCode { get; set; } = null!;

    public DateTime InsertDate { get; set; }

    public string InsertUser { get; set; } = null!;

    public DateTime UpdateDate { get; set; }

    public string UpdateUser { get; set; } = null!;

    public string? Mail { get; set; }

    public string? TaxCode { get; set; }

    public virtual ICollection<Agency> Agency { get; set; } = new List<Agency>();

    public virtual Configuration? Configuration { get; set; }

    public virtual ICollection<Customer> Customer { get; set; } = new List<Customer>();

    public virtual ICollection<CustomerRequiredField> CustomerRequiredField { get; set; } = new List<CustomerRequiredField>();

    public virtual ICollection<Principal> Principal { get; set; } = new List<Principal>();

    public virtual ICollection<Product> Product { get; set; } = new List<Product>();

    public virtual ICollection<Region> Region { get; set; } = new List<Region>();
}

File: Cc.Ui/src/models/company.ts
import { Entity, resource, field, required, association } from '../framework/entity';
import { Pagination, Pager } from '../framework/core/actions';
// import { CompanyDocument } from './companyDocument';

@resource('company')
export class Company extends Entity {

    @field()
    id: number;
    @field()
    insertDate: string;
    @field()
    insertUser: string;
    @field()
    updateDate: string;
    @field()
    updateUser: string;
    @field()
    businessName: string;
    @field()
    vatCode: string;
    @field()
    address: string;
    @field()
    taxCode: string;
    @field()
    mail: string;


    constructor(data?: any) {
        super()
        this.init(data);      
    }

    static newItem(): Company {

        let item = new Company();
        return item;
    }
}

@resource('company')
export class CompanySearchModel extends Entity implements Pagination {
    
    @field()
    companyId?: number;
    @field()
    businessName?: string;
    @field()
    vatCode?: string;

    @field()
    pager: Pager = <Pager> {
        take: 100,
        orderBy: 'id'
    };
}